testservice/sdk_client_entity.go

```diff
@@ -384,6 +394,59 @@ func makeSDKConfig(config servicedef.SDKConfigParams, sdkLog ldlog.Loggers) ld.C
 			builder.PayloadFilter(config.Polling.Filter.String())
 		}
 		ret.DataSource = builder
+	} else if config.ServiceEndpoints == nil {
+		ret.DataSource = ldcomponents.ExternalUpdatesOnly()
+	}
+
+	if config.PersistentDataStore != nil {
+		var builder *ldcomponents.PersistentDataStoreBuilder
+		switch config.PersistentDataStore.Store.Type {
+		case servicedef.Redis:
+			dsBuilder := ldredis.DataStore().URL(config.PersistentDataStore.Store.DSN)
+			if config.PersistentDataStore.Store.Prefix != nil {
+				dsBuilder.Prefix(*config.PersistentDataStore.Store.Prefix)
+			}
+			builder = ldcomponents.PersistentDataStore(dsBuilder)
+		case servicedef.Consul:
+			dsBuilder := ldconsul.DataStore().Address(config.PersistentDataStore.Store.DSN)
+			if config.PersistentDataStore.Store.Prefix != nil {
+				dsBuilder.Prefix(*config.PersistentDataStore.Store.Prefix)
+			}
+			builder = ldcomponents.PersistentDataStore(dsBuilder)
+		case servicedef.DynamoDB:
+			cfg, err := awsconfig.LoadDefaultConfig(context.Background(),
+				awsconfig.WithRegion("us-east-1"),
+				awsconfig.WithCredentialsProvider(credentials.NewStaticCredentialsProvider("dummy", "dummy", "dummy")),
+			)
+			if err != nil {
```

Comment:
The function that calls `makeSDKConfig` is capable of returning an error, so it would be good to change the signature here so we can bubble this back up

Reply:
Done.